src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/ManagedX509ExtensionProcessor.cs

```diff
@@ -225,9 +226,21 @@ public virtual byte[] ComputeCapiSha1OfPublicKey(PublicKey key)
             AsnWriter writer = new AsnWriter(AsnEncodingRules.DER);
             spki.Encode(writer);
 
-            using (SHA1 hash = SHA1.Create())
+            byte[] rented = CryptoPool.Rent(writer.GetEncodedLength());
+
+            try
+            {
+                if (!writer.TryEncode(rented, out int bytesWritten))
```

Comment:
@bartonjs at some point... somewhere.. there was a thought about a `SpanAction`-like API that could give the underlying bytes to avoid a copy. Did that ever materialize?

Reply:
It was part of the API proposal, but got removed.  (Also, turns out that SpanAction isn't available in netstandard2.0, so it would complicate things)